### PR TITLE
fix: fixes version lock to set fact

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
fix version lock issue to use a fact instead of a local register

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by joejulian — APPROVED** [review]

File: ansible/roles/setup_versionlock/tasks/redhat.yaml [hunk 1]
```diff
@@ -34,3 +34,7 @@
     warn: false
   ignore_errors: true
   changed_when: false
+
+- name: Export versionlocklist
```
Comment (dkoshkin) [inline ansible/roles/setup_versionlock/tasks/redhat.yaml:39]:
```suggestion
- name: export versionlocklist
```

File: ansible/roles/setup_versionlock/tasks/suse.yaml [hunk 2]
```diff
@@ -6,3 +6,7 @@
     warn: false
   ignore_errors: true
   changed_when: false
+
+- name: Export versionlocklist
```
Comment (dkoshkin) [inline ansible/roles/setup_versionlock/tasks/suse.yaml:10]:
```suggestion
- name: export versionlocklist
```

File: ansible/provision.yaml [hunk 3]
```diff
@@ -30,6 +30,7 @@
       when: ansible_os_family == "Flatcar"
   roles:
     - role: offline
+    - role: setup_versionlock
```
Comment (dkoshkin) [inline ansible/provision.yaml:33]: Do we need both this and https://github.com/mesosphere/konvoy-image-builder/search?q=setup_versionlock?

Comment (faiq) [reply ansible/provision.yaml:33]: honestly im not sure. let me run this again and figure it out.

**Review by dkoshkin — APPROVED** [review]

**Comment by faiq** [comment]
It looks like this nvidia stuff is also failing on main: https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2eCentOS7nvidia/3401014

**Pull request merged**